### ISSUE TITLE
How to choose the appropriate training set resolution? image/label: non-normalized or out of bounds coordinates

### ISSUE THREAD

**Issue by su322** [ultralytics/ultralytics#19591]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

我找了一个数据集的图片尺寸是[1914,1052]，按照默认训练参数的imgsz=640，出现了下面这样的警告，但是我确定我的标签是正确归一化了的。我推测是尺寸方面的问题导致的，但是我不太懂。我想知道我有什么办法可以消除这种警告，或者有没有什么推荐的原图尺寸呢，我可以直接改原图尺寸并重新打标签。

I found a dataset with image sizes of [1914, 1052], and according to the default training parameter imgsz=640, the following warning appeared, but I am certain that my labels have been correctly normalized. I speculate it's due to size issues, but I don't quite understand. I would like to know if there is any way I can eliminate this warning, or if there are any recommended original image sizes. I can simply change the original image size and re label it.

```
train: WARNING ⚠️ D:\project\project_py\ultralytics-main\datasets\images\train\3398351.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0301      1.2992]
```

### Additional
train.py
```
from ultralytics.models import YOLO
import os

os.environ['KMP_DUPLICATE_LIB_OK'] = 'True'

if __name__ == '__main__':
    model = YOLO(model='ultralytics/cfg/models/11/yolo11.yaml')
    model.load('yolo11n.pt')
    model.train(data='./data.yaml', epochs=2, batch=1, device='cpu', imgsz=640, workers=2, cache=False,
                amp=True, mosaic=False, rect=True, project='runs/train', name='exp')
```
data.yaml
```
 # dataset path
train: ./images/train
val: ./images/test
test: ./images/test

# number of classes
nc: 1

# class names
names: ['car']

```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @su322, thank you for reaching out and for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for helpful information on training and usage. For your specific query, you might find the [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) guide particularly useful.

From the warning message you provided, it does appear that there might be an issue with the image/label coordinate normalization. To assist you better, if this is a 🐛 Bug Report, we kindly ask you to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can debug effectively. This would include:

1. A small dataset or a single image and label file causing the issue.
2. The exact command you used to execute the training.
3. Any custom configuration or parameters used, if applicable.

If this is a ❓ Question related to dataset preparation or image resolution, please provide additional details about your dataset, such as the labeling format, any preprocessing steps applied, and whether you've verified the labels against the image dimensions. This will help us provide more precise guidance.

To eliminate potential software-related issues, please ensure you're using the latest version of `ultralytics`. You can upgrade it as follows:

```bash
pip install -U ultralytics
```

You can also test your training in any of the following verified environments to rule out system-specific problems:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

For real-time support, you can also join our [Discord](https://discord.com/invite/ultralytics) community 🎧, engage in discussions on [Discourse](https://community.ultralytics.com/), or connect with fellow users on our [Subreddit](https://reddit.com/r/Ultralytics).

An Ultralytics engineer will review your issue soon and provide further assistance. Thank you for your patience and for helping us improve! 😊

**Comment by Y-T-G** [comment]
It's not a problem with imgsz but your labels. If they're normalized correctly, they shouldn't be above `1.0`

**Comment by su322** [comment]
No, I'm sure my labels are fine, but it's possible that I used someone else's code to convert XML format to TXT format and I don't know what happened. I have decided to remake the dataset, thank you.

**Comment by glenn-jocher** [comment]
@su322 great to hear you're proactively addressing the dataset issue! For your label conversion, ensure coordinates are normalized by **original image width/height** (1914x1052) rather than target `imgsz=640`. The YOLO Label Formats guide explains the required format where all coordinates must be 0-1 normalized. If you need to validate future labels, our Data Preprocessing Techniques page has helpful normalization checks. 

Let us know if you encounter any other issues during your dataset recreation! 🚀
